src/main/java/com/eternitywall/ots/BitcoinNode.java

```diff
@@ -83,51 +82,58 @@ public BlockHeader getBlockHeader(Integer height) throws Exception {
     }
 
     public BlockHeader getBlockHeader(String hash) throws Exception {
-        if(hash==null)
-            return null;
+        if (hash == null) {
+            return null;    // TODO: This is not nice - it creates harder to diagnose NPEs later down the call chain
+        }
+
         JSONObject json = new JSONObject();
         json.put("id", "java");
         json.put("method", "getblockheader");
-        JSONArray array=new JSONArray();
+        JSONArray array = new JSONArray();
         array.put(hash);
         json.put("params", array);
         JSONObject jsonObject = callRPC(json);
+
         BlockHeader blockHeader = new BlockHeader();
         JSONObject result = jsonObject.getJSONObject("result");
         blockHeader.setMerkleroot(result.getString("merkleroot"));
         blockHeader.setBlockHash(hash);
         blockHeader.setTime(String.valueOf(result.getInt("time")));
+
         return blockHeader;
     }
 
     public String getBlockHash(Integer height) throws Exception {
         JSONObject json = new JSONObject();
         json.put("id", "java");
         json.put("method", "getblockhash");
-        JSONArray array=new JSONArray();
+        JSONArray array = new JSONArray();
         array.put(height);
         json.put("params", array);
         JSONObject jsonObject = callRPC(json);
-        if(jsonObject==null)
+
+        if (jsonObject == null) {
             return null;
+        }
+
         return jsonObject.getString("result");
     }
 
     private JSONObject callRPC(JSONObject query) throws Exception {
-
-        String s=query.toString();
+        String s = query.toString();
         URL url = new URL(urlString);
         Request request = new Request(url);
-        Map<String,String> headers = new HashMap<>();
+        Map<String, String> headers = new HashMap<>();
         headers.put("Authorization", "Basic " + authString);
         request.setHeaders(headers);
         request.setData(s.getBytes());
-        Response response = null;  //sync call
+        Response response = null;  // sync call TODO: What?
```

Comment:
checked history, was in a try catch
no reason to keep the null init and the comment